src/Components/CardList.tsx

```diff
@@ -80,14 +90,88 @@ export const CardList = ({
     if (item?.format && formats.includes(item?.format)) return true
   }
 
+  function genderFilter(
+    item: ServicePreview,
+    genderInput: Array<string> | undefined
+  ) {
+    // If no preference is set for user or service, return everything
+    if (genderInput?.length === 0 || item?.gender?.length === 0) {
+      return true
+    }
+
+    genderInput?.filter((gen) => {
+      const gender = gen as Gender
+      if (!item?.gender?.includes(gender)) {
+        return false
+      } else {
+        return true
+      }
```

Comment:
```ts
return item?.gender?.includes(gender)
```

(you've done similar things in a few places in this file, but I won't comment the rest)